samples/tetris/build.gradle.kts

```diff
@@ -1,42 +1,79 @@
 import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinNativeTarget
-import org.jetbrains.kotlin.gradle.plugin.mpp.NativeBuildType
 
 plugins {
     kotlin("multiplatform")
 }
 
-val hostOs = System.getProperty("os.name")
-val isWindows = hostOs.startsWith("Windows")
-
-// If RaspberryPi target is activated.
-val isRaspberryPiBuild =
-    project.findProperty("tetris.raspberrypi.build")?.toString()?.toBoolean() == true
-
-// If host platform is Windows and x86 target is activated.
-val isMingwX86Build =
-    isWindows && project.findProperty("tetris.mingwX86.build")?.toString()?.toBoolean() == true
-
-val winCompiledResourceFile = buildDir.resolve("compiledWindowsResources/Tetris.res")
-
 val kotlinNativeDataPath = System.getenv("KONAN_DATA_DIR")?.let { File(it) }
     ?: File(System.getProperty("user.home")).resolve(".konan")
-
-val mingwPath = if (isMingwX86Build)
-    File(System.getenv("MINGW32_DIR") ?: "C:/msys32/mingw32")
-else
-    File(System.getenv("MINGW64_DIR") ?: "C:/msys64/mingw64")
+val mingw64Path = File(System.getenv("MINGW64_DIR") ?: "C:/msys64/mingw64")
+val mingw32Path = File(System.getenv("MINGW32_DIR") ?: "C:/msys64/mingw32")
 
 kotlin {
-    createRequestedTarget("tetris").apply {
+    val hostOs = System.getProperty("os.name")
+    if (hostOs == "Mac OS X") {
+        macosX64()
+    }
+    if (hostOs == "Linux") {
+        linuxX64()
+    }
+    if (hostOs.startsWith("Windows")) {
+        mingwX64()
+        mingwX86()
+    }
+    linuxArm32Hfp()
+
+    targets.withType<KotlinNativeTarget> {
+        sourceSets["${targetName}Main"].apply {
+            kotlin.srcDir("src/tetrisMain/kotlin")
+        }
+
         binaries {
             executable {
                 entryPoint = "sample.tetris.main"
+
+                // Compile Windows Resources
+                if (konanTarget.family == org.jetbrains.kotlin.konan.target.Family.MINGW) {
```

Comment:
`KonanTarget` is an internal thing. It does appear in buildscripts of some our libraries but it's better to minimize its usage. Could you check `preset` instead of `konanTarget` here and below?

Reply:
Done